implementation/src/main/java/io/smallrye/mutiny/operators/multi/builders/BufferItemMultiEmitter.java

```diff
@@ -30,7 +33,7 @@ public MultiEmitter<T> emit(T t) {
             fail(new NullPointerException("`emit` called with `null`."));
             return this;
         }
-        if (queue.offer(t)) {
+        if (queue.offer(t) && !Queues.isOverflowing(queue, overflowBufferSize)) {
```

Comment:
you can better enforce the limits here by using a utility which wrap 
[offerIfBelowThreshold](https://github.com/search?q=repo%3AJCTools%2FJCTools%20offerIfBelowThreshold&type=code), while available: if not available and appealing please fill a JCTools issue and I can provide it for the other missing queues

Reply:
The main thing here is that the rest of the code base assumes `Queue<T>`, so that method isn't (easily) available.

Reply:
Moved to per-operator tracking